### ISSUE TITLE
regex-shorthand: Expected regexp to be of type `string`, got `object`

### ISSUE THREAD

**Issue by junqdu** [sindresorhus/eslint-plugin-unicorn#149]
```

$(npm bin)/eslint . --fix
The rule `angular/service-name` will be split up to different rules in the next version. Please read the docs for more information
Expected regexp to be of type `string`, got `object`
TypeError: Expected regexp to be of type `string`, got `object`
    at module.exports (.../node_modules/clean-regexp/index.js:18:9)
    at NewExpression[callee.name="RegExp"] (.../node_modules/eslint-plugin-unicorn/rules/regex-shorthand.js:32:23)
    at listeners.(anonymous function).forEach.listener (.../node_modules/eslint/lib/util/safe-emitter.js:47:58)
    at Array.forEach (native)
    at Object.emit (.../node_modules/eslint/lib/util/safe-emitter.js:47:38)
    at NodeEventGenerator.applySelector (.../node_modules/eslint/lib/util/node-event-generator.js:251:26)
    at NodeEventGenerator.applySelectors (.../node_modules/eslint/lib/util/node-event-generator.js:280:22)
    at NodeEventGenerator.enterNode (.../node_modules/eslint/lib/util/node-event-generator.js:294:14)
    at CodePathAnalyzer.enterNode (.../node_modules/eslint/lib/code-path-analysis/code-path-analyzer.js:608:23)
    at Traverser.enter [as _enter] (.../node_modules/eslint/lib/linter.js:1006:32)
    at Traverser._traverse (.../node_modules/eslint/lib/util/traverser.js:132:14)
    at Traverser._traverse (.../node_modules/eslint/lib/util/traverser.js:147:30)
    at Traverser._traverse (.../node_modules/eslint/lib/util/traverser.js:144:34)
    at Traverser._traverse (.../node_modules/eslint/lib/util/traverser.js:147:30)
    at Traverser._traverse (.../node_modules/eslint/lib/util/traverser.js:144:34)
    at Traverser._traverse (.../node_modules/eslint/lib/util/traverser.js:144:34)
```

**Comment by SamVerschueren** [comment]
Any idea what line in your code is failing? Is it possible to copy-paste it? Maybe I missed an edge case.

**Comment by junqdu** [comment]
@SamVerschueren I think it is something like `new RegExp(/^[^*]*[*]?$/)` caused the error.

**Comment by sindresorhus** [comment]
@junqdu Nope, it's not that. It passes the tests: https://github.com/sindresorhus/eslint-plugin-unicorn/commit/bdebf03e3b8e1f7de44184c93f1e6098c396d2bd

Instead of guessing, can you reduce your code to the minimum amount that still makes it fail and share that here?

**Comment by junqdu** [comment]
@sindresorhus 
I setup an angular tutorial app https://github.com/junqdu/eslintest
Error shows up at https://github.com/junqdu/eslintest/blob/master/app/core/checkmark/checkmark.filter.js#L7

![screen shot 2018-02-11 at 1 48 24 am](https://user-images.githubusercontent.com/4250728/36072074-f348ccfe-0ecd-11e8-8a36-9c5310cbee3a.png)

**Comment by sindresorhus** [comment]
The following runs fine for me without crashing:

```js
'use strict';

angular.
  module('core').
  filter('checkmark', function() {
    return function(input) {
      const regex = new RegExp(/^[^*]*[*]?$/);
      return input ? '\u2713' : '\u2718';
    };
  });
```

**Comment by sindresorhus** [comment]
I'm happy to look further into it if you submit a [failing test](https://github.com/avajs/ava#failing-tests).

**Comment by junqdu** [comment]
@sindresorhus seems to be fixed in 4.0.1
